patrol/patrol_test.go

```diff
@@ -0,0 +1,60 @@
+package patrol_test
+
+import (
+	"reflect"
+	"sync"
+	"testing"
+	"time"
+
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+
+	"github.com/Fresh-Tracks/bomb-squad/patrol"
+	"github.com/Fresh-Tracks/bomb-squad/util"
+)
+
+func TestPatrol(t *testing.T) {
+	client, err := util.HttpClient()
+	Must(t, err)
+
+	wg := sync.WaitGroup{}
+
+	s := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(200)
+		w.Write([]byte(`{"status":"success","data":{"resultType":"vector","result":[]}}`))
+
+		AssertEquals(t, "/api/v1/query?query=topk%280%2Cdelta%28card_count%5B1m%5D%29%29", r.RequestURI)
+
+		wg.Done()
+	}))
+	defer s.Close()
+
+	promurl, err := url.Parse(s.URL)
+	Must(t, err)
+
+	p := patrol.Patrol{
+		Client:   client,
+		PromURL:  promurl,
+		Interval: 100 * time.Millisecond,
+	}
+
+	wg.Add(1)
+	go func() {
+		p.Run()
+	}()
+	wg.Wait()
+
+}
+
+func Must(t *testing.T, err error) {
```

Comment:
I've seen some nice conveniences using `github.com/stretchr/testify/assert` and `github.com/stretchr/testify/require` on other projects. Basically just a set of convenience functions like `require.NoError(t, err)` which mimics what you've written here. There are a _ton_ of them, too, so it saves a bunch of time reimplementing these funcs ourselves. :-)

Not urgent for this PR, just wanted to mention it for future use. I'll likely be pulling those pkgs in for whatever tests I write when I get some time, I can refactor these then if needed.

Reply:
Yeah, we can add it, I generally run away from introducing dependencies for as long as I can, but down the line, it will be necessary.

Reply:
No rush. I'll get to it at some point once I figure out some other bits related to testing K8s interactions. For now it's good as-is.